enroll/models.py

```diff
@@ -1,3 +1,80 @@
 from django.db import models
+from django.contrib.auth.models import AbstractUser
 
-# Create your models here.
+from enroll.fields import DayOfTheWeekField, DAY_OF_THE_WEEK
+from enroll.validators import validate_by_user_type
+from enroll.types import UserType
+from enroll.utils import time_plus_minutes
+
+
+class User(AbstractUser):
+    user_type = models.PositiveSmallIntegerField(choices=UserType().get_choices())
+
+    def __str__(self):
+        return super().__str__() + ' (' + (UserType.get_by_key(self.user_type)) + ')'
+
+
+class Lecturer(models.Model):
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+    account = models.OneToOneField(
+        User,
+        on_delete=models.CASCADE,
+    )
+
+    def clean(self):
+        validate_by_user_type('teacher')(self.account)
+
+    def __str__(self):
+        return f'{self.first_name} {self.last_name}'
+
+
+class Course(models.Model):
+    code = models.CharField(max_length=40)
+    full_name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.full_name
+
+
+class ClassTime(models.Model):
```

Comment:
Consider adding frequency of classes - each week, every 2 weeks (A or B), irregular.